Bluepill-runner/Bluepill-runner/BPRunner.m

```diff
@@ -81,6 +86,129 @@ + (instancetype)BPRunnerWithConfig:(BPConfiguration *)config
+        [self.testHostForSimUDID setValue:simulatorUDIDString forKey:self.config.appBundlePath];
+        [BPUtils printInfo:INFO withString:@"created sim template: %@ for app host: %@", simulatorUDIDString, self.config.appBundlePath];
+    } else {
+        // This is for testing in command line when we pass the xctestrun file
+        NSMutableSet *hostBundles = [[NSMutableSet alloc] init];
+        for (BPXCTestFile* bundle in testBundles) {
+            [hostBundles addObject:bundle.testHostPath];
+        }
+        if ([testBundles count] == 0) {
+            [BPUtils printInfo:ERROR withString:@"no host bundle founnd!"];
+        }
+        for (NSString *appPath in hostBundles) {
+            NSError *error = nil;
+            simulatorUDIDString = [self installApplicationwithHost:appPath withError:error];
+            [BPUtils printInfo:INFO withString:@"created sim template: %@ for app host: %@", simulatorUDIDString, appPath];
+            [self.testHostForSimUDID setValue:simulatorUDIDString forKey:appPath];
+        }
+    }
+    return;
+}
+
+- (NSString *)installApplicationwithHost:(NSString *)testHost withError:(NSError *)error {
+    SimServiceContext *sc = [SimServiceContext sharedServiceContextForDeveloperDir:self.config.xcodePath error:&error];
+    if (!sc) {
+        [BPUtils printInfo:ERROR withString:[NSString stringWithFormat:@"SimServiceContext failed: %@", [error localizedDescription]]];
+        exit(1);
+    }
+    SimDeviceSet *deviceSet = [sc defaultDeviceSetWithError:&error];
+    if (!deviceSet) {
+        [BPUtils printInfo:ERROR withString:[NSString stringWithFormat:@"SimDeviceSet failed: %@", [error localizedDescription]]];
+        exit(1);
+    }
+    
+    self.simDeviceType = nil;
+    for (SimDeviceType *type in [sc supportedDeviceTypes]) {
+        if ([[type name] isEqualToString:@BP_DEFAULT_DEVICE_TYPE]) {
+            self.simDeviceType = type;
+            break;
+        }
+    }
+    
+    if (!self.simDeviceType) {
+        [BPUtils printInfo:ERROR withString:@"Invalid Device Type!"];
+        exit(1);
+    }
+    
+    self.simRuntime = nil;
+    for (SimRuntime *runtime in [sc supportedRuntimes]) {
+        if ([[runtime name] isEqualToString:self.config.runtime]) {
+            self.simRuntime = runtime;
+            break;
+        }
+    }
+    
+    if (!self.simRuntime) {
+        [BPUtils printInfo:ERROR withString:@"Invalid runtime Type!"];
+        exit(1);
+    }
+    
+    SimDevice *simDevice = [deviceSet createDeviceWithType:self.simDeviceType
+                                                   runtime:self.simRuntime
+                                                      name:@"BP_Simultator_Blueprint"
+                                                     error:&error];
+    [self.simDeviceTemplates addObject:simDevice];
+    if (!simDevice || error) {
+        [BPUtils printInfo:ERROR withString:@"create simulator failed with error: %@", error];
```

Comment:
Please properly capitalize all INFO and ERROR messages.

Reply:
done